backend/src/api/FeedController.ts

```diff
@@ -70,11 +70,15 @@ export default class FeedController {
 
         try {
             const sorting = await this.userManager.getFeedSorting(userId, 1);
-            // const total = await this.feedManager.getSubscriptionsTotal(userId);
+            const restrictedPosts = await this.feedManager.getRestrictedPosts(userId, page, perPage, format, sorting);
+
             const {
                 total,
                 posts: rawPosts
-            } = await this.feedManager.getSubscriptionFeed(userId, page, perPage, format, sorting);
+            } = restrictedPosts ? {
+                total: await this.postManager.getPostsByUserTotal(userId),
+                posts: restrictedPosts
+            } : await this.feedManager.getSubscriptionFeed(userId, page, perPage, format, sorting);
```

Comment:
General question for this PR - what is instead explicitely branching for restricte user content we embed restrictions into all manager methods? In this way we can be sure that there is no case missed that still fetches all the posts without restrictions.

Reply:
Technically possible, although not consistent with the current approach (restrictions are already checked in controllers), also requires a bit more effort/refactoring, e.g. pair methods like `FeedManager.getAllPosts` / `FeedManager.getAllPostsTotal` would have to either calculate restrictions twice, or be joined into a single method.

Don't think it's worth the effort atm / could be refactored later if there is a valid reason to do so.

Reply:
I see your point but it is still easy to miss something with this approach - for example `Post#translate` API is not limited and can be used to retrieve any comment or post currently